torchvision/utils.py

```diff
@@ -198,45 +199,50 @@ def draw_bounding_boxes(
     elif image.size(0) not in {1, 3}:
         raise ValueError("Only grayscale and RGB images are supported")
 
+    num_boxes = boxes.shape[0]
+
+    if labels is None:
+        labels = [None] * num_boxes
```

Comment:
I'm going to need @pmeier 's enlightment on the mypy issue here https://app.circleci.com/pipelines/github/pytorch/vision/14562/workflows/da348bb7-e7cb-46f5-9d6e-0ea7248d59f1/jobs/1171625

What would you suggest?

I tried

```py
        labels: Union[List[str], List[None]] = [None] * num_boxes
```

But `mypy` complains with 

```
torchvision/utils.py:205: error: Name "labels" already defined on line 154  [no-redef]
            labels: Union[List[str], List[None]] = [None] * num_boxes
            ^
```


despite the `allow_redefinition = True` option.

Reply:
Very naturally:

```py
        labels: Union[List[str], List[None]] = [None] * num_boxes  # type: ignore[no-redef]
```

won't work either because mypy will instead use the first (outdated) definition instead of this new one.

Reply:
> despite the `allow_redefinition = True` option.

This happens because `allow_redefinition` only applies to cases where the value was used at least once. In your case, we overwrite `labels` without usage and thus this option has no effect. 

> What would you suggest?

If `labels = [None] * ...` is a valid value, I would simply change the input annotation to reflect that: `Optional[List[Optional[str]]]`. This also gives the user the option to exclude certain labels (`labels=["label1", None, "label2"]`) and should be compliant with the rest of the code.

Reply:
Thanks @pmeier for the info.

I'm not sure I understand what the "nesting rule" is meant to enforce. For example this makes mypy happy and yet it's strictly equivalent to our code:

```py
    labels: Union[List[str], List[None]] = [None] * num_boxes if labels is None else labels
```

I'd prefer not allowing None values in the `labels` list for users, as I'm not sure there's a use-case where users would have labels for some boxes, but not all (worst cast scenario they can always pass empty  strings). In fact, before this PR, passing a `None`  value for a label would fail as `draw.text` doesn't support it.


So I'll just silence mypy here (twice), because this error  doesn't make much sense.